File: python/api_cong.js
'use strict';

var request = require('request');

function getKeyPhrasesWrapper (rawInput) {
    return getKeyPhrases (toJson (rawInput));
}

function getKeyPhrases (json) {
    // Azure portal URL.
    const base_url = 'https://westus.api.cognitive.microsoft.com/';

    const headers = {
        'Content-Type':'application/json', 
        'Accept': 'application/json',
        'Ocp-Apim-Subscription-Key':'db9b5fd1cb7c493789373ebf27c0524d'};

    const num_languages = 1;

    const batch_keyphrase_url = base_url + 'text/analytics/v2.0/keyPhrases';

    var options = {
        'url': batch_keyphrase_url,
        'method': 'POST',
        'headers': headers,
        'data': {"documents": JSON.stringify(json)}
    }
    
    console.log(JSON.stringify(json));
    request(options, function (error, response, body) {
        console.log(body);
    })
}


function extratSentences ( line ) {
    let sentences = line.split('.');
    const length = sentences.length;

    for ( let i = 0; i < length; i++ ) {
        sentences[i] = sentences[i].replace('"', '');
    }

    return sentences;
}

function toJson (rawInput) {
    // Converts raw input into json with each sentence as an item in an array
    let attributeCount = 1;
    // Each line is treated as one paragraph
    // Each paragraph call api once

    let json = {
        "documents": []
    };

    let lines = rawInput.split('\n');

    lines.forEach ( function(line) {
        let sentences = extratSentences(line);
        sentences.forEach ( function ( sentence ) {
            if ( sentence !== '' ) {
                let attribute = {
                    "language": "en",
                    "id": ""+attributeCount,
                    "text": sentence
                };
                json.documents.push(attribute);
                attributeCount++;
            }
        });
    });

    return json;
}

getKeyPhrasesWrapper("Hello. My name is Domingo.\nHow are you?");

File: python/api_cong.py

# Simple program that demonstrates how to invoke Azure ML Text Analytics API: key phrases, language and sentiment detection.
import urllib2
import urllib
import sys
import base64
import json


# Azure portal URL.
base_url = 'https://westus.api.cognitive.microsoft.com/'

headers = {'Content-Type':'application/json', 'Ocp-Apim-Subscription-Key':'db9b5fd1cb7c493789373ebf27c0524d'}
      
# Path to file with JSON inputs.
file_path = './text.json'
f = open(file_path, 'r')
input_texts = f.read()

num_detect_langs = 1;


# Detect key phrases.
batch_keyphrase_url = base_url + 'text/analytics/v2.0/keyPhrases'
req = urllib2.Request(batch_keyphrase_url, input_texts, headers) 
response = urllib2.urlopen(req)
result = response.read()
obj = json.loads(result)
for keyphrase_analysis in obj['documents']:
    print('Key phrases ' + str(keyphrase_analysis['id']) + ': ' + ', '.join(map(str,keyphrase_analysis['keyPhrases'])))


